crates/consensus/src/validation.rs

```diff
@@ -471,6 +499,22 @@ mod tests {
         }
     }
 
+    impl WithdrawalsProvider for Provider {
+        fn latest_withdrawal(
```

Comment:
for test, the current code like this sees can only mock one kind of case? I need test both None and Some, any existing approach in reth codebase?